curiosrc/market/deal_ingest.go

```diff
@@ -129,10 +267,250 @@ func (p *PieceIngester) AllocatePieceToSector(ctx context.Context, maddr address
 		return api.SectorOffset{}, xerrors.Errorf("expected one sector number")
 	}
 
-	// After we insert the piece/sector_pipeline entries, the lpseal/poller will take it from here
+	if p.sealRightNow {
+		err = p.SectorStartSealing(ctx, num[0])
+		if err != nil {
+			return api.SectorOffset{}, xerrors.Errorf("SectorStartSealing: %w", err)
+		}
+	}
 
 	return api.SectorOffset{
 		Sector: num[0],
 		Offset: 0,
 	}, nil
 }
+
+func (p *PieceIngester) allocateToExisting(ctx context.Context, piece lpiece.PieceDealInfo, rawSize int64, source url.URL, dataHdrJson, propJson []byte) (bool, api.SectorOffset, error) {
+
+	var ret api.SectorOffset
+	var allocated bool
+	var rerr error
+
+	comm, err := p.db.BeginTransaction(ctx, func(tx *harmonydb.Tx) (commit bool, err error) {
+		openSectors, err := p.getOpenSectors(tx)
+		if err != nil {
+			return false, err
+		}
+
+		pieceSize := piece.Size()
+		for _, sec := range openSectors {
+			sec := sec
+			if sec.currentSize+pieceSize < abi.PaddedPieceSize(p.sectorSize) {
+
+				ret.Sector = sec.number
+				ret.Offset = sec.currentSize + 1 // Current filled up size + 1
+
+				// Insert market deal to DB for the sector
+				if piece.DealProposal != nil {
+					for {
+						cn, err := tx.Exec(`SELECT insert_sector_market_piece($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14)`,
+							p.mid, sec.number, sec.index+1,
+							piece.DealProposal.PieceCID, piece.DealProposal.PieceSize,
+							source.String(), dataHdrJson, rawSize, !piece.KeepUnsealed,
+							piece.PublishCid, piece.DealID, propJson, piece.DealSchedule.StartEpoch, piece.DealSchedule.EndEpoch)
+
+						if err != nil {
+							var pqErr *pq.Error
+							ok := errors.As(err, &pqErr)
+							if ok && pqErr.Code == "P0001" {
```

Comment:
* What is P0001?
* I'm quite sure that our db code returns `pgx.Err..` errors (from yugabyte pgx, the non-YB driver errors will NOT match)

Reply:
https://www.postgresql.org/docs/8.2/errcodes-appendix.html This is how PGSQL raises an exception. It is just an exception code. This is not specific to YB. 
I hope I haven't over-engineered with the retry coupled with Transaction atomicity.

Reply:
I can test this exception manually in our table just to be sure.

Reply:
Yes, but I'm pretty sure that yb-pgx errors don't cast to pq.Error

Reply:
You were right. It doesn't work. I have removed this error handling and retry. It doesn't make sense with transaction anyway. It was an overkill.